content/post/095-road-to-dht.md

```diff
@@ -0,0 +1,84 @@
+---
+date: 2020-05-19
+url: 2020-05-19-road-to-dht
+title: The Road to the new DHT
+author: Adin Schmahmann
+header_image: 095-road-to-dht.png
+tags: go-ipfs, 0.5 release
+---
+
+![The Road to the new DHT](/header_images/095-road-to-dht.png)
+
+At the end of April, we released our largest update to go-ipfs to date: [IPFS 0.5.0](https://blog.ipfs.io/2020-04-28-go-ipfs-0-5-0/). This upgrade brings major performance and reliability improvements to IPFS — especially on the content discovery and routing front. These gains are brought to you largely by our rework of the Distributed Hash Table (DHT) which powers most peer and content discovery in the IPFS Public Network.
+
+We’d like to take you through our journey to _re-write_ the DHT - from identifying the underlying issues, to how we approached designing a solution, to ensuring it would **work** in practice. Our work led to major performance gains, as well as a new development process and [Testground](https://github.com/testground/testground), a whole new tool for large-scale network testing [(read more)](https://blog.ipfs.io/2020-05-06-launching-testground/). 
+
+**If you want to make use of these new improvements, please [upgrade to the latest IPFS now](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure/#use-ipfs-update)!**
+<br />
+
+![Brendan quote](/095-road-to-dht/Brendan-quote.png)
+
+## First a refresher:  What’s the DHT?
+
+The [DHT, or distributed hash table](https://docs.ipfs.io/guides/concepts/dht/), acts like both a catalog and navigation system for IPFS, helping the network keep track of and find data. A regular hash table is a key-value store where the keys are hashed and stored in one place. In the case of IPFS, the keys are the Content Identifiers (CIDs) of a block of data, and the values are the set of peers who have each block.
+
+The IPFS DHT distributes and stores these pairs of keys and values in small tables across many nodes or peers throughout the network. Peers, in turn, store information about where to physically locate specific content. Kademlia, our DHT implementation, helps keep track of which nodes are _providing_ what data to others in the network.
```

Comment:
@momack2 while this gives a good general idea the distinction I was wondering if we care about is that while for finding/providing IPFS content the keys are CIDs and values are provider records in go/js-ipfs we store other types of data in the DHT other than just provider records (e.g. peer records and IPNS records).

The first paragraph is actually totally correct on its own, but the second paragraph which has words like "The IPFS DHT" and "our DHT implementation" technically blurs the lines a little. While maybe there are some slightly better words here it's probably fine for this type of post.

Reply:
```suggestion
The [DHT, or distributed hash table](https://docs.ipfs.io/guides/concepts/dht/), acts like both a catalog and navigation system for IPFS, helping the network keep track of and find data. A regular hash table is a key-value store where the keys are hashed and stored in one place. In the case of IPFS, the keys are the Content Identifiers (CIDs) of a block of data, and the values are the set of peers who have each block.

The DHT distributes and stores these pairs of keys and values in small tables across many nodes or peers throughout the network. Peers, in turn, store information about where to physically locate specific content. Kademlia, our DHT implementation, helps keep track of which nodes are _providing_ what data to others in the network.
```
Gotcha. Changed the first line, but kept the rest. These statements don't rule out that other types of data may be served as well.